src/mono/mono/utils/mono-mmap-wasm.c

```diff
@@ -72,6 +72,13 @@ prot_from_flags (int flags)
 		prot |= PROT_WRITE;
 	if (flags & MONO_MMAP_EXEC)
 		prot |= PROT_EXEC;
+	
+#if HOST_WASI
+	// The mmap in wasi-sdk rejects PROT_NONE, but otherwise disregards the flags
+	// We just need to pass an acceptable value
+	prot |= PROT_READ;
```

Comment:
It might be better to just 'return PROT_READ' at the top of this function when HOST_WASI to make it clearer what's up

Reply:
Good suggestion - done.